clang/lib/Driver/ToolChains/Clang.cpp

```diff
@@ -4119,20 +4119,26 @@ void Clang::ConstructJob(Compilation &C, const JobAction &JA,
     }
   }
 
+  Arg *SYCLStdArg = Args.getLastArg(options::OPT_sycl_std_EQ);
+
     if (IsSYCL) {
-    if (Arg *A = Args.getLastArg(options::OPT_sycl_std_EQ)) {
-      A->render(Args, CmdArgs);
+    if (SYCLStdArg) {
+      SYCLStdArg->render(Args, CmdArgs);
       CmdArgs.push_back("-fsycl-std-layout-kernel-params");
     } else {
       // Ensure the default version in SYCL mode is 1.2.1 (aka 2017)
       CmdArgs.push_back("-sycl-std=2017");
-      // The user had not pass SYCL version, thus we'll employ no-sycl-strict
-      // to allow address-space unqualified pointers in function params/return
-      // along with marking the same function with explicit SYCL_EXTERNAL
-      CmdArgs.push_back("-Wno-sycl-strict");
     }
     }
 
+  if ((Args.hasArg(options::OPT_fsycl_device_only) ||
+       Args.hasArg(options::OPT_fsycl) || IsSYCL) && !SYCLStdArg) {
```

Comment:
```suggestion
  if (UseSYCLTriple && !SYCLStdArg) {
```
...as I don't imagine a scenario where SYCL offloading actions would be enabled w/o `-fsycl`

Reply:
AFAIU there is no offloading upon `-fsycl -fsycl-device-only`

Reply:
Indeed, but offloading scenarios are included in your original condition (`IsSYCL`). Anyhow, the alternative I've suggested seems to be covering all relevant scenarios.

Reply:
Done